asynciterator.ts

```diff
@@ -1252,6 +1254,135 @@ function destinationFillBuffer<S>(this: InternalSource<S>) {
 }
 
 
+export abstract class SynchronousTransformIterator<S, D = S> extends AsyncIterator<D> {
```

Comment:
```suggestion
export abstract class MappedIterator<S, D = S> extends AsyncIterator<D> {
```

Let's just call it this.

The fact that the `Synchronous` pertains to `Transform` but not to `Iterator` is too hard to understand.